src/libpriv/rpmostree-core.c

```diff
@@ -1172,6 +1172,12 @@ rpmostree_context_download_metadata (RpmOstreeContext *self,
                      NULL);
   }
 
+  /* Don't try to keep multiple kernels per root; that's a traditional thing,
+   * ostree binds kernel + userspace.
+   */
+  dnf_sack_set_installonly (dnf_context_get_sack (self->dnfctx), NULL);
+  dnf_sack_set_installonly_limit (dnf_context_get_sack (self->dnfctx), 0);
```

Comment:
Hmm, let's put this in `prepare` instead? Feels more natural to be in the same path as `dnf_goal_depsolve`.